src/main/java/com/abdullah/coding/challenge/services/AdminService.java

```diff
@@ -0,0 +1,67 @@
+package com.abdullah.coding.challenge.services;
+
+import com.abdullah.coding.challenge.Response;
+import com.abdullah.coding.challenge.dto.CabDTO;
+import com.abdullah.coding.challenge.entities.Booking;
+import com.abdullah.coding.challenge.entities.Cab;
+import com.abdullah.coding.challenge.enums.BookingStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+public class AdminService {
+    @Autowired
+    private BookingService bookingService;
+    @Autowired
+    private CabService cabService;
+    @Autowired
+    private CustomerService customerService;
+
+    public Response<?> cancelBooking(String bookingCode) {
+        return bookingService.cancelBooking(bookingCode);
+    }
+
+    @Transactional
+    public Response<?> assignCab(int bookingId) {
+
+        Response<Booking> bookingResponse = bookingService.getBooking(bookingId);
+        if (bookingResponse.getStatusCode() == HttpStatus.OK) {
```

Comment:
It's because the booking service is aware of it's responsibilities. I did not want to replicate same logic in this method. Although it is possible for this particular instance. But what if in the future, our business logic becomes complex to get a booking? Does that mean I would have to introduce that code complexity to this method too or just let the booking service handle the complexity and act as an interface where any other service can interact.?